Adds the search code API

zamith opened the pull request:
The sort and order parameters are optional, so there is one function
that takes those arguments and one that doesn't.

edgurgel commented on lib/tentacat.ex:
```diff
@@ -56,9 +56,9 @@ defmodule Tentacat do
   end
 
-  def get(path, client, params \\ []) do
+  def get(path, client, params \\ [], encode \\ true) do
```
Can we make this an option as a keyword list?

Something like

```def get(path, client, params \\ [], options \\ [])

Keyword.get(options, :encode, true)
```

It would be clearer than just passing true/false

edgurgel commented:
I'm sorry for the **massive** delay!

zamith commented:
Apparently GitHub API has changed, so I updated the examples, and now I don't remember why I needed the non encoded url, so I just removed that code. I can put it back in if you want.

duksis commented:
works on my machine :) and looks good to me - thanks! @zamith
